udata/linkchecker/checker.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals, absolute_import
+
+from datetime import datetime, timedelta
+
+from flask import current_app
+
+from .backends import get as get_linkchecker
+
+
+def _get_check_keys(the_dict):
+    return {k: v for k, v in the_dict.iteritems() if k.startswith('check:')}
+
+
+def get_cache(resource):
+    '''Return a cached version from a resource's check if any and fresh'''
+    if resource.extras.get('check:date'):
+        cache_duration = current_app.config['LINKCHECKING_CACHE_DURATION']
+        if resource.extras['check:date'] <= datetime.now() \
+                + timedelta(seconds=cache_duration):
+            status = resource.extras.get('check:status')
+            return _get_check_keys(resource.extras), status
+
+
+def check_resource(resource):
+    '''
+    Check a resource availability against a linkchecker backend
+
+    The linkchecker used can be configured on a resource basis by setting
+    the `resource.extras['check:checker']` attribute with a key that points
+    to a valid `udata.linkcheckers` entrypoint. If not set, it will
+    fallback on croquemort. If set to `no_check` it will assume the resource
+    is available via a dummy linkchecker.
+    '''
+    cached_check = get_cache(resource)
+    if cached_check:
+        return cached_check
+    linkchecker_type = resource.extras.get('check:checker', 'croquemort')
+    LinkChecker = get_linkchecker(linkchecker_type)
+    if not LinkChecker:
+        return {'error': 'No linkchecker configured.'}, 503
```

Comment:
Maybe return `unkown` instead of an error. If it's optionnal, we might want not to consider the lack of linkchecker as an error.
Alternative: ensure there is always default one like the dummy one.

Reply:
There will always be the dummy one unless you configure a bad `LINKCHECKING_DEFAULT_LINKCHECKER`, and in this case it make sense to raise an error. 

NB: `503` is not really considered as an error on the frontend for now, nothing happens.